cpp/src/decisiontree/levelalgo/levelfunc_classifier.cuh

```diff
@@ -99,40 +112,49 @@ void grow_deep_tree_classification(
       "Max node limit reached. Requested nodes %d > %d max nodes at depth %d\n",
       n_nodes, tempmem->max_nodes_per_level, depth);
 
+    update_feature_sampling(h_colids, d_colids, h_colstart, d_colstart, Ncols,
+                            ncols_sampled, n_nodes, mtg, dist, feature_selector,
+                            tempmem);
     get_histogram_classification(data, labels, flagsptr, sample_cnt, nrows,
-                                 ncols, n_unique_labels, nbins, n_nodes,
-                                 split_algo, tempmem, d_histogram);
+                                 Ncols, ncols_sampled, n_unique_labels, nbins,
+                                 n_nodes, split_algo, tempmem, d_histogram);
 
     float* infogain = tempmem->h_outgain->data();
     if (split_cr == ML::CRITERION::GINI) {
       get_best_split_classification<T, GiniFunctor, GiniDevFunctor>(
-        h_histogram, d_histogram, feature_selector, d_colids, nbins,
-        n_unique_labels, n_nodes, depth, min_rows_per_node, split_algo,
-        infogain, sparse_histstate, sparsetree, sparsesize, sparse_nodelist,
-        h_split_colidx, h_split_binidx, d_split_colidx, d_split_binidx,
-        tempmem);
+        h_histogram, d_histogram, h_colids, d_colids, h_colstart, d_colstart,
+        Ncols, ncols_sampled, nbins, n_unique_labels, n_nodes, depth,
+        min_rows_per_node, split_algo, infogain, h_parent_hist, h_child_hist,
```

Comment:
As a more general thing (for a future change, not this one!) the number of parameters in RF internal functions keeps growing. It would be worth thinking about a struct to capture this internal state used in splits.

Reply:
yes i want to do a clean up. I will file a cleanup PR after this.